testHelperFixtures/src/test/java/org/opensearch/migrations/testutils/NettyLeakCheckTestExtensionTest.java

```diff
@@ -26,18 +26,21 @@ public void beforeTest() {
 
     @AfterEach
     public void afterTest() {
-        var observedTestDuration = Duration.ofNanos(System.nanoTime()-startTimeNanos);
-        switch (testCase) {
-            case "testMaxTimeSupercedesReps":
-                Assertions.assertTrue(counter < 20, "counter=" + counter);
-                Assertions.assertTrue(Duration.ofMillis(100).minus(observedTestDuration).isNegative());
-                break;
-            case "testMinTimeSupercedesReps":
-                Assertions.assertTrue(counter > 1, "counter=" + counter);
-                Assertions.assertTrue(Duration.ofMillis(100).minus(observedTestDuration).isNegative());
-                break;
-            default:
-                Assertions.fail("unknown test case: " + testCase);
+        // testCase may be null if extension skips beforeEach
+        if (testCase != null) {
```

Comment:
Please invert the flow of control to reduce nesting, `if (testCase == null) { return; }`